temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingWorkflowOutboundCallsInterceptor.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.opentracing.internal;
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+import io.temporal.common.interceptors.Header;
+import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptor;
+import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptorBase;
+import io.temporal.opentracing.OpenTracingOptions;
+import io.temporal.opentracing.SpanOperationType;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowInfo;
+
+public class OpenTracingWorkflowOutboundCallsInterceptor
+    extends WorkflowOutboundCallsInterceptorBase {
+  private final OpenTracingOptions openTracingOptions;
+
+  public OpenTracingWorkflowOutboundCallsInterceptor(
+      WorkflowOutboundCallsInterceptor next, OpenTracingOptions openTracingOptions) {
+    super(next);
+    this.openTracingOptions = openTracingOptions;
+  }
+
+  @Override
+  public <R> ActivityOutput<R> executeActivity(ActivityInput<R> input) {
+    createAndPassActivityStartSpan(input.getActivityName(), input.getHeader());
```

Comment:
I would change it to a span that wraps activity execution, not only its invocation. This comment applies to all other wrapped APIs.

Reply:
For wrapping the activity execution with have a RunActivity span. It wraps the execution. This is a StartActivity span. It just basically shows that the activity was triggered/scheduled. I think we should close the span either immediately or after invocation to show in the trace that it happened. It can be especially important for WorkflowStart span of long-living activity. You don't want to wait till the end of a workflow to publish a root span.